rust/src/nfs/nfs2_records.rs

```diff
@@ -96,3 +96,211 @@ pub fn parse_nfs2_attribs(i: &[u8]) -> IResult<&[u8], Nfs2Attributes> {
+
+        // packet_bytes
+        let buf: &[u8] = &[
+
+        // [file_handle]
+            0x00, 0x10, 0x10, 0x85, 0x00, 0x00, 0x03, 0xe7,
+            0x00, 0x0a, 0x00, 0x00, 0x00, 0x00, 0xb2, 0x5a,
+            0x00, 0x00, 0x00, 0x29, 0x00, 0x0a, 0x00, 0x00,
+            0x00, 0x00, 0xb2, 0x5a, 0x00, 0x00, 0x00, 0x29,
+
+        // [name]
+        // name_len
+            0x00, 0x00, 0x00, 0x02,
+
+        // name_contents: (am)
+            0x61, 0x6d,
+
+        // _name_padding
+            0x00, 0x00,
+        ];
+
+        let (_, handle) = parse_nfs2_handle(buf).expect("Parsing nfs2_handle failed!");
+
+        let result = parse_nfs2_request_lookup(buf);
+        match result {
+            Ok((r, request)) => {
+                assert_eq!(r.len(), 0);
+                assert_eq!(request.handle, handle);
+                assert_eq!(request.name_vec, b"am".to_vec());
+            }
+            Err(error) => { panic!("Parsing failed: {:?}", error); }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_request_read() {
+
+        // packet_bytes
+        let buf: &[u8] = &[
+
+        // [file_handle]
+            0x00, 0x10, 0x10, 0x85, 0x00, 0x00, 0x03, 0xe7,
+            0x00, 0x0a, 0x00, 0x00, 0x00, 0x00, 0xb2, 0x5d,
+            0x00, 0x00, 0x00, 0x2a, 0x00, 0x0a, 0x00, 0x00,
+            0x00, 0x00, 0xb2, 0x5a, 0x00, 0x00, 0x00, 0x29,
+
+        // offset
+            0x00, 0x00, 0x00, 0x00,
+        // count
+            0x00, 0x00, 0x20, 0x00,
+        // total count
+            0x00, 0x00, 0x20, 0x00
+        ];
+
+        let (_, handle) = parse_nfs2_handle(buf).expect("Parsing nfs2_handle failed!");
+
+        let result = parse_nfs2_request_read(buf);
+        match result {
+            Ok((r, request)) => {
+                assert_eq!(r.len(), 4);
+                assert_eq!(request.handle, handle);
+                assert_eq!(request.offset, 0);
+            }
+            Err(error) => { panic!("Parsing failed: {:?}", error); }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_reply_read() {
+
+        let buf: &[u8] = &[
+
+        // Status: NFS_OK - (0)
+            0x00, 0x00, 0x00, 0x00,
+
+        // attr_blob
+            0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x81, 0xa4,
+            0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x0b,
+            0x00, 0x00, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x02, 0x00, 0x10, 0x10, 0x85,
+            0x00, 0x00, 0xb2, 0x5d, 0x38, 0x47, 0x75, 0xea,
+            0x00, 0x0b, 0x71, 0xb0, 0x38, 0x47, 0x71, 0xc4,
+            0x00, 0x08, 0xb2, 0x90, 0x38, 0x47, 0x75, 0xea,
+            0x00, 0x09, 0x00, 0xb0,
+
+        // data_len: (11)
+            0x00, 0x00, 0x00, 0x0b,
+
+        // data_contents: ("the b file")
+            0x74, 0x68, 0x65, 0x20,
+            0x62, 0x20, 0x66, 0x69, 0x6c, 0x65, 0x0a,
+
+        // _data_padding: parsed as part of data_contents
+            0x00,
```

Comment:
I'm not sure about breaking down the data like this. I think the con is too many newlines and lines in general. But, a big pro is legible fields. I'll let Victor or Jason decide this.

Reply:
I like it.  Can help when you're debugging a test someone else wrote and you're not that familiar with the contents, as you generally can't just paste this into wireshark to get the break down.